src/esp32/esp32_eth.c

```diff
@@ -126,6 +137,12 @@ bool mgos_ethernet_init(void) {
     return false;
   }
 
+  ret = esp32_set_host_name(mgos_sys_config_get_eth());
```

Comment:
I would set host name only if it is explicitely set in the configuration:

```c
if (mgos_sys_config_get_eth_dhcp_hostname() != NULL &&
    esp_netif_set_hostname(esp_netif_get_handle_from_ifkey("ETH_DEF"), 
      mgos_sys_config_get_eth_dhcp_hostname()) != ESP_OK) {
  LOG(LL_ERROR, ("ETH: Failed to set host name"));
  goto out;
}
```

Reply:
Should be fixed now, thanks for the advice. https://github.com/mongoose-os-libs/ethernet/compare/master...ALLTERCO:master

Reply:
i think the function can be inlined since it's only used once. you already have the netif handle in `eth_if` here